src/test/file/shd-test-file.c

```diff
@@ -15,188 +17,137 @@
 #include <sys/uio.h>
 #include <errno.h>
 
-static int _test_newfile() {
-    FILE* file = fopen("testfile", "w");
-    if(file == NULL) {
-        fprintf(stdout, "error: could not create new file\n");
-        return -1;
-    }
+// Similar to g_assert_true, but include stringified errno on failure.
+#define assert_true_errno(c)                                                   \
+    if (!(c)) {                                                                \
+        g_error("!(%s): %s", #c, strerror(errno));                             \
+        g_test_fail();                                                         \
+    }
+
+#define assert_nonnull_errno(p) assert_true_errno((p) != NULL)
+#define assert_nonneg_errno(p) assert_true_errno((p) >= 0)
+
+// Assert that errno is the expected value.
+#define assert_errno_is(e)                                                     \
+    {                                                                          \
+        int _errno = errno;                                                    \
+        int _e = e;                                                            \
+        if (_e != _errno) {                                                    \
+            g_error("Got errno %d (%s) instead of %d (%s)", _errno,            \
+                    strerror(_errno), _e, strerror(_e));                       \
+            g_test_fail();                                                     \
+        }                                                                      \
+    }
```

Comment:
I'm guessing some of this would be useful if we continue converting other tests to the glibtest framework, but I suppose we can refactor it to a common file then.

Reply:
Yeah, I was thinking of waiting until they're needed again, but I agree it seems likely if we continue using glib testing.